src/App.js

```diff
@@ -3,51 +3,123 @@ import axios from 'axios';
 import MovieListRenderer from './components/movie'
 import Navbar from './components/navbar';
 import ShowMsg from './components/showmsg';
+import Login from './components/login';
+import fire from './config/'
 
 class App extends Component {
   constructor(props) {
     super(props)
     // state
     this.state = {
+      user: null,
       error : null,
       isLoaded : false,
       movies : [],
       liked: [],
       query: '',
-      filtered: []
+      filtered: [],
+      pageNumber: 1,
     }
   }
 
   componentDidMount(){
-    axios.get(`${process.env.REACT_APP_MOVIE_API_KEY}`)
-      .then((result) => {
-        this.setState({
-          movies : result.data.results,
-          isLoaded : true,
-        });
-      }).catch((error) => {
+    this.authLestiner();
+    this.fetchMovies();
+    document.addEventListener('scroll', this.trackScrolling);
+  }
+  authLestiner = () => {
+    fire.auth().onAuthStateChanged((user) => {
+      user ?
         this.setState({
-          error
+          user
         })
-      });
+      :
+      this.setState({
+        user: null
+      })
+    })
+  }
+  handleLogIn = () => {
+    const email = document.querySelector('#email').value
+    const password = document.querySelector('#password').value
+    fire.auth().signInWithEmailAndPassword(email, password)
+    .then((user) =>{
+      this.setState({
+        user
+      })
+    })
+    .catch((error) =>{
+      console.log(error.toString())
+    })
   }
-  
+
+  handleSignUp = () => {
+    const email = document.querySelector('#email').value
```

Comment:
handle that with react state, don't use document
revision using form with react